spec/controllers/songs_controller_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe SongsController do
+
+  before :each do
+      @song = FactoryGirl.create(:song)
+  end
+
+  it { should use_before_filter(:find_song) }
+
+
+  describe 'GET #edit' do
+
+    it 'assigns requested song to @song' do
+      get :edit, id: @song
+      expect(assigns(:song)).to eq(@song)
+    end
+
+    it 'render the :edit template' do
+      get :edit, id: @song
+      expect(response).to render_template :edit
+    end
+  end
+
+
+  describe 'PUT update' do
+
+    before do
+      @song.update_attributes(artist: 'Frodo')
+    end
+
+    it 'assign the requested song to @song' do
+      put :update, id: @song
```

Comment:
What exactly do you update here? All updates were performed in ``` before ``` block in ``` update_attributes ``` isn't it?